monai/data/meta_tensor.py

```diff
@@ -479,10 +479,19 @@ def peek_pending_shape(self):
         return tuple(convert_to_numpy(self.shape, wrap_sequence=True).tolist()[1:]) if res is None else res
 
     def peek_pending_affine(self):
-        res = None
-        if self.pending_operations:
-            res = self.pending_operations[-1].get(LazyAttr.AFFINE, None)
-        return self.affine if res is None else res
+        res = self.affine
+        for p in self.pending_operations:
+            next_matrix = convert_to_tensor(p.get(LazyAttr.AFFINE))
+            if next_matrix is None:
+                continue
+            res = convert_to_dst_type(res, next_matrix)[0]
+            res = monai.transforms.lazy.utils.combine_transforms(res, next_matrix)
+        return res
+
+    def peek_pending_rank(self):
+        a = self.pending_operations[-1].get(LazyAttr.AFFINE, None) if self.pending_operations else self.affine
+        r = max(1, len(a) - 1)
+        return convert_to_dst_type(r, self.affine)[0]
```

Comment:
Do we have a special reason to convert this rank to "dst type"( maybe tensor) which may cause an error when we use this rank?

Reply:
this is a good question, currently the affine can be float32/float64, so we need `peek_pending_affine_dtype`. in practice I think it's more convenient to merge `peek_pending_affine_dtype` into `peek_pending_rank`, because both functions are mostly used together...

thinking about this again, maybe we should always use float64 for the affine matrix, float32 is not precise enough in most cases. I'll update this PR to use float64 affine.

Reply:
added a new commit to use double for the affine https://github.com/Project-MONAI/MONAI/pull/5860/commits/4a46fda6b44a6ccdb200d2cb94cb3ed50f30667b